internal/cmd/login.go

```diff
@@ -176,25 +179,19 @@ func loginToInfra(cli *CLI, client *api.Client, loginReq *api.LoginRequest) erro
 			return err
 		}
 
-		client.AccessKey = loginRes.AccessKey
-		if _, err := client.UpdateUser(&api.UpdateUserRequest{ID: loginRes.UserID, Password: password}); err != nil {
+		lc.APIClient.AccessKey = loginRes.AccessKey
+		if _, err := lc.APIClient.UpdateUser(&api.UpdateUserRequest{ID: loginRes.UserID, Password: password}); err != nil {
 			return err
 		}
 
 		fmt.Fprintf(os.Stderr, "  Updated password.\n")
 	}
 
-	if err := updateInfraConfig(client, loginReq, loginRes); err != nil {
+	if err := updateInfraConfig(lc, loginReq, loginRes); err != nil {
 		return err
 	}
 
-	// Client needs to be refreshed from here onwards, based on the newly saved infra configuration.
-	client, err = defaultAPIClient()
-	if err != nil {
-		return err
-	}
```

Comment:
I think I remember this being done because the call to get the user's grants in `updateKubeConfig` was not getting the new access key that was issued on login. It might have been fixed as a side-effect somewhere else.

Reply:
I was very confused by this, because the existing tests were passing. I found an incorrect `return nil` in `updateKubeConfig` which was masking the problem. With that fixed the existing tests fail.

I fixed this by setting the access key on the API client once login is successful.

Reply:
Great catch